### ISSUE TITLE
`DIRNAME` and `FILENAME` global variables?

### ISSUE THREAD

**Issue by danielbayley** [yaml/yamlscript#199]
Node has `__dirname` and `__filename` global constants, or their modern ESM eqivalent `import.meta.dirname`/`.filename`. Would it be worth adding similar, alongside `FILE`, to the [existing global variables](https://github.com/yaml/yamlscript/blob/main/doc/cheat.md?plain=1#L185)?

**Comment by ingydotnet** [comment]
I was actually about to add that this morning while working on https://github.com/ingydotnet/advent-of-code/

I needed to read a data file that sat next to the script.

I had
```
defn main():
  data =: slurp('data.txt')
```
but that only worked when I ran the script from its directory.

I then changed it to 
```
defn main():
  data =: slurp("$fs-dirname(FILE)/data.txt")
```

which worked but seemed like a common need. wanted:
```
defn main():
  data =: slurp("$DIR/data.txt")
```

but ended up doing
```
defn main(file='./data.txt'):
  data = file:slurp
```

So yeah, I'll add it today :)

**Comment by ingydotnet** [comment]
I assume that's what you meant...

**Comment by ingydotnet** [comment]
Sorry, been distracted with family and holidays and advent of code.

I'll try again tomorrow.

https://github.com/ingydotnet/advent-of-code/tree/main/2024 FYI :)

**Comment by danielbayley** [comment]
> Sorry, been distracted with family and holidays

@ingydotnet No worries, totally relatable!

**Comment by ingydotnet** [comment]
@danielbayley Added 010fb2f5 to devel branch.

**Comment by ingydotnet** [comment]
Released in 0.1.87
